### PR TITLE
highs: init at 1.6.0

### PR DESCRIPTION
**Pull request by david-r-cox**
## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [x] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by quantenzitrone** [comment]
You need to add yourself to `maintainers/maintainer-list.nix` in a separate commit.

**Comment by david-r-cox** [comment]
> You need to add yourself to `maintainers/maintainer-list.nix` (in a separate commit).

Thanks @quantenzitrone -- I updated `maintainer-list.nix` in https://github.com/NixOS/nixpkgs/pull/273252 and will wait for that to merge before running `nixpkgs-review`.

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:15]:
```suggestion
    hash = "sha256-Wa5ivUJk0t58FhZD0zy0zxHHj4/p8e9WcxXwu5zenxI=";
```

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:7]:
```suggestion
stdenv.mkDerivation rec {
```

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:26]: please set `meta.mainProgram` if there's a main binary

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:23]:
```suggestion
    description = "High performance serial and parallel linear solver";
```
per nixpkgs manual, package name shouldn't be in desc

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
As far as the maintainer commit, can you rebase this branch to come off 6290195a26e1e7e0d13d51df3fb8baeba4152845 where you added yourself to maintainers? That way you're not really duplicating work, but ofborg can eval it.

**Comment by david-r-cox** [comment]
Result of `nixpkgs-review pr 273289` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>highs</li>
  </ul>
</details>

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:21]:
```suggestion

  doCheck = true;
 
```

probably good to enable tests here just as a sanity check

**Review by eclairevoyant — CHANGES_REQUESTED** [review]
One more bit of feedback, but otherwise tested out on my end, seems good

File: pkgs/by-name/hi/highs/package.nix
Comment (david-r-cox) [reply pkgs/by-name/hi/highs/package.nix:21]: Good call, thanks @eclairevoyant

**Review by eclairevoyant — APPROVED** [review]

**Review by quantenzitrone — APPROVED** [review]

**Comment by david-r-cox** [comment]
@eclairevoyant is there anything else that I should be doing on this PR? This (along with #273296, #273295, and #273252) is my first time contributing to nixpkgs. Thanks again for the review.

**Comment by eclairevoyant** [comment]
I approved but I don't have access to merge

File: pkgs/by-name/hi/highs/package.nix
Comment (eclairevoyant) [inline pkgs/by-name/hi/highs/package.nix:29]: Looking at the CI again - do you know why the tests are failing on darwin?

**Comment by KiaraGrouwstra** [comment]
looks like https://github.com/NixOS/nixpkgs/pull/295386 already got merged

**Pull request closed without merging**
